keyless/pepper/service/src/lib.rs

```diff
@@ -357,13 +358,25 @@ async fn process_common(
         .kid
         .ok_or_else(|| BadRequest("missing kid in JWT".to_string()))?;
 
-    let sig_pub_key = jwk::cached_decoding_key(&claims.claims.iss, &key_id)
+    let cached_key = jwk::cached_decoding_key_as_rsa(&claims.claims.iss, &key_id);
+
+    let jwk;
+    match cached_key {
+        Ok(key) => jwk = key,
+        Err(_) => {
+            jwk = get_federated_jwk(&jwt)
```

Comment:
But don't you wanna save the key returned by `get_federated_jwk` in the cache and expire it after a while?

Reply:
Otherwise, you will be triggering a new request repeatedly for the same federated `iss`...

Reply:
is federated jwk caching worth it?

some reasons to not cache:
- implementation and test complexity
- a lot of cache miss when e.g. we set cache capacity 100 but receive traffic involving 200 tenants (assuming we use a LRU cache with customizable capacity)

Reply:
I guess we'll see in production if we have issues without caching?

Some thoughts:
- The implementation complexity is small (e.g., see the cache code I use for keyless Poseidon hashes)
- I suppose you do need to write some tests to make sure that the cache hits and that it gets cleared, yes.
- I think you could set the cache to be quite large (e.g., 2 KiB * 100,000 =~ 195 MiB)

Reply:
I figured it was not worth caching for now.  In the future I think it is worthwhile to cache but I figure the performance impact will be not terribly significant.

I vote to revisit.